src/main/scala/com/databricks/labs/mosaic/sql/extensions/MosaicGDAL.scala

```diff
@@ -0,0 +1,36 @@
+package com.databricks.labs.mosaic.sql.extensions
+
+import com.databricks.labs.mosaic.gdal.MosaicGDAL
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.SparkSessionExtensions
+
+/**
+  * Supports automatic registration of SQL expressions at the cluster start up
+  * time. This class registers and injects a rule that based on the configured
+  * index system and configured geometry api adds corresponding SQL expressions
+  * to spark session. These rules are activated just after the spark session has
+  * been created. If unsupported pair of index system and geometry api has been
+  * provided this operation will crash the code in order to make sure correct
+  * functionality has been enabled.
+  */
+class MosaicGDAL extends (SparkSessionExtensions => Unit) with Logging {
+
+    /**
+      * Constructor for the MosaicSQL extension. All the registration logic
+      * happens before a nop rule has been injected.
+      * @param ext
+      *   Hook to spark session that is needed for expression registration.
+      */
+    override def apply(ext: SparkSessionExtensions): Unit = {
+        ext.injectCheckRule(spark => {
+            val enableGDAL = spark.conf.get("spark.mosaic.gdal.native", "false").toBoolean
```

Comment:
What does "native" mean in the context of `spark.mosaic.gdal.native`?

Reply:
if to use mosaic to install, or GDAL is already installed.
I will remove it since we have not tested it with arbitrary GDAL versions - so it is a bit pointless to allow for something that can go any way and turn out super unstable.